src/doc/src/reference/build-scripts.md

```diff
@@ -233,7 +235,9 @@ equivalent to using [`rustc-link-lib`](#rustc-link-lib) and
 
 The `rustc-cfg` instruction tells Cargo to pass the given value to the
 [`--cfg` flag][option-cfg] to the compiler. This may be used for compile-time
-detection of features to enable [conditional compilation].
+detection of features to enable [conditional compilation]. Each custom cfgs
+must **always** be declared using the [`cargo::rustc-check-cfg`](#rustc-check-cfg)
+instruction.
```

Comment:
Changed to:

> Custom cfgs must either be expected using the [`cargo::rustc-check-cfg`](#rustc-check-cfg) instruction or usage will need to allow the `unexpected_cfgs` lint to avoid unexpected cfgs warnings.